feat: add metrics components to frontend

sconnole opened the pull request:
The original PR has stalled https://github.com/cachethq/core/pull/58 and I would like to be able to use metrics in my cachet application. This is more recently rebased from main than the original PR

here the metrics chart rendering (which is all that I need for my application)
![image](https://github.com/user-attachments/assets/390aac4f-32c1-4180-a679-34fdf51cefcf)

Happy to make updates as requested. @jbrooksuk

jbrooksuk commented:
Great start on this, @sconnole. There are a few things we need to sort before we can merge:

1. Please don't commit the compiled files, these will be generated automatically once merged.
2. The chart text isn't visible in light mode. I tried a quick pass at supporting light / dark mode with CSS variables and I can get it to work, but it's slow updating the colors when toggling.
3. Speaking of colors, we probably should use the accent colors for the chart itself.
4. We need more padding around the chart container. It's a bit tight at the moment.

sconnole commented:
@jbrooksuk Good call on light theme. I forgot to QA that. I have updated the following 

1. removed compiled files
2. Updated to show labels in light mode 
![image](https://github.com/user-attachments/assets/eb341ce6-4872-43a7-b99b-3eaaf9af8529)
![image](https://github.com/user-attachments/assets/da63386a-ae75-4ef3-97e7-f99309f5ff70)
3. Added tailwind based accent colors for the graph
4. Adding padding-3 to the chart

sconnole commented:
🎉  Glad it's merged in. Thanks @Ionys320 for getting the building block going!

jbrooksuk commented:
Awesome team work 🫶
